src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -721,6 +741,8 @@ MulticopterPositionControl::task_main()
 			_att_sp.disable_mc_yaw_control = false;
 			_att_sp.apply_flaps = false;
 
+			_wv_controller.update(matrix::Quatf(&_att_sp.q_d[0]), _local_pos.yaw);
```

Comment:
You can build a matrix::Quatf with a 4 element float array directly rather than the address of the first element. 

`matrix::Quatf(_att_sp.q_d)`

It's quite minor, but eventually all of these memory safety holes can be closed.